pkg/omf/functions/index/omf.index.query.fish

```diff
@@ -0,0 +1,102 @@
+function omf.index.query -d 'Query packages in the index'
+  set -q XDG_CACHE_HOME
+    and set -l index_path "$XDG_CACHE_HOME/omf"
+    or set -l index_path "$HOME/.cache/omf"
+
+  set -l q_type any
+  set -l q_name ''
+  set -l q_text ''
+
+  # Parse search terms.
+  for arg in $argv
+    switch "$arg"
+      case 'type=a*'
+        set q_type any
+      case 'type=p*'
+        set q_type package
+      case 'type=t*'
+        set q_type theme
+      case 'name=?*'
+        set -l IFS '='
+        echo "$arg" | read dummy q_name
+      case 'text=?*'
+        set -l IFS '='
+        echo "$arg" | read dummy q_text
+      case '*'
+        echo "Invalid search term: '$arg'" >&2
+        return 1
+    end
+  end
+
+  # Determine what files to search against based on type.
+  switch $q_type
+    case package
+      set packages $index_path/*/packages/*
+    case theme
+      set packages $index_path/*/themes/*
+    case '*'
+      set packages $index_path/*/{packages,themes}/*
+  end
+
+  # If there are no packages, there is nothing to search.
+  if not set -q packages[1]
+    return 1
+  end
+
+  # Perform a text search if any textual terms were given.
+  if test -n "$q_name" -o -n "$q_text"
+    set results (command awk -v "q_name=$q_name" -v "q_text=$q_text" '
+      function flush() {
+        if (package && name_matches && text_matches) {
```

Comment:
When you put it that way, I don't see much of a need for the `name` filter...

Currently most of these can be done with the function though:

1. `omf.index.query --text=pattern` searches the name and any property value.
2. Restricting to specific property can't be done.
3. `omf.index.query --name=pattern`
4. `omf.index.query --name=pattern1 --text=pattern2`

Note that the default patterns for both `text` and `name` is `''`, which matches everything.